app/file.go

```diff
@@ -217,15 +217,27 @@ func (s *Server) removeFile(path string) *model.AppError {
 }
 
 func (a *App) ListDirectory(path string) ([]string, *model.AppError) {
-	return a.Srv().listDirectory(path)
+	return a.Srv().listDirectory(path, false)
 }
 
-func (s *Server) listDirectory(path string) ([]string, *model.AppError) {
+func (a *App) ListDirectoryRecursively(path string) ([]string, *model.AppError) {
```

Comment:
You now added this ListDirectoryRecursively, but looks like you are not using it anywhere, I guess you need to add this to the place where you were trying to fix the lack of recursion problem.

Reply:
yes, I have to add it to the Enterprise repo, will try to do it today, sadly I get other priorities in the way and I'm not moving as fast as I'd like